R/AbstractAnnData.R

```diff
@@ -71,7 +71,9 @@ AbstractAnnData <- R6::R6Class("AbstractAnnData", # nolint
       )) {
         attr_key <- paste0(attribute, "_keys")
         if (!is.null(self[[attr_key]])) {
-          cat("    ", pretty_print(attribute, self[[attr_key]]()), "\n", sep = "")
+          if (length(self[[attr_key]]()) > 0) {
+            cat("    ", pretty_print(attribute, self[[attr_key]]()), "\n", sep = "")
+          }
```

Comment:
Perhaps only call the function once

```suggestion
          slot_keys <- self[[attr_key]]()
          if (length(slot_keys) > 0) {
            cat("    ", pretty_print(attribute, slot_keys, "\n", sep = "")
```
          }